address_book/internal/service/service.go

```diff
@@ -21,7 +25,30 @@ func (a *AddressBookService) AddContact(_ context.Context, in *pb.AddContactRequ
 		}, nil
 	}
 
-	a.data.Store(in.Contact.Phone, in.Contact)
+	var count int64 = 0
+	if err := a.db.Limit(1).Find(&models.Contact{}, in.Contact.Phone).Count(&count).Error; err != nil {
+		return &pb.AddContactResponse{
+			Msg: "adding error",
```

Comment:
Here and below: please don't use hard-coded messages, use constants or variables instead.

Reply:
Done